pallets/utility/src/benchmarking.rs

```diff
@@ -0,0 +1,178 @@
+use crate::*;
+use pallet_balances::{self as balances, Call as BalancesCall};
+use pallet_identity::benchmarking::User;
+
+use frame_benchmarking::benchmarks;
+use sp_runtime::{traits::StaticLookup, MultiSignature};
+
+#[cfg(not(feature = "std"))]
+use hex_literal::hex;
+#[cfg(feature = "std")]
+use sp_core::sr25519::Signature;
+
+const MAX_CALLS: u32 = 30;
+
+fn make_calls<T: Trait>(c: u32) -> Vec<<T as Trait>::Call> {
+    let call: <T as Trait>::Call = frame_system::Call::<T>::remark(vec![]).into();
+    vec![call; c as usize]
+}
+
+fn make_transfer_calls<T: Trait>(
+    c: u32,
+    to: T::AccountId,
+    amount: u128,
+) -> Vec<<T as Trait>::Call> {
+    let idx = <T as frame_system::Trait>::Lookup::unlookup(to);
+    let call: <T as Trait>::Call = BalancesCall::transfer(idx, amount.into()).into();
+    vec![call; c as usize]
+}
+
+fn verify_free_balance<T: Trait>(account: &T::AccountId, expected_balance: u128) {
+    let acc_balance = balances::Module::<T>::free_balance(account);
+    assert_eq!(acc_balance, expected_balance.into())
+}
+
+struct RelayTxSetup<T: Trait> {
+    pub caller: T::AccountId,
+    pub target: T::AccountId,
+    pub signature: T::OffChainSignature,
+    pub call: UniqueCall<<T as Trait>::Call>,
+}
+
+/// Relay transaction setup for native code.
+/// It prepares a relayed transfer from `bob` to `alice` which is going to be called by `alice`.
+/// In order to do that, `bob` has to sign the call. We can sign only in native execution, because
+/// WASM does not enable `full_crypto` feature.
```

Comment:
I wonder if we can enable `full_crypto` on wasm in `runtime-benchmarks` flag.

Reply:
It adds bloat so it's not recommended for normal releases but I don't think there's any problem in using it for benchmarks.